api/validator.go

```diff
@@ -0,0 +1,143 @@
+package api
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+type address struct {
+	Host string
+	Port int
+}
+
+func (a address) String() string {
+	return fmt.Sprintf("%s:%d", a.Host, a.Port)
+}
+
+func (r Ingress) IsValid(cloudProvider string) error {
+	addrs := make(map[address]int)
+	nodePorts := make(map[int]int)
+	for ri, rule := range r.Spec.Rules {
+		if rule.HTTP != nil {
+			addr := address{Host: rule.Host}
+			if port, err := checkRequiredPort(rule.HTTP.Port); err != nil {
```

Comment:
what if i do not set any port value. this should set 80/443 by default for http, as it currently is. this should not through an error.